pkg/util/miseadapter/authorizer.go

```diff
@@ -0,0 +1,79 @@
+package miseadapter
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type MISEAdapter interface {
+	IsAuthorized(ctx context.Context, r *http.Request) (bool, error)
+	IsReady() bool
+}
+
+type miseAdapter struct {
+	client *Client
+	log    *logrus.Entry
+}
+
+func NewAuthorizer(miseAddress string, log *logrus.Entry) MISEAdapter {
```

Comment:
At first I thought this was authorizing something, but it looks like it's just authenticating using the credentials retrieved from the Authorization header. Correct me if I'm wrong here. 

The term authenticated and authorized have been loosely used here, it doesn't tell immediately whether it's actually authorizing something, therefore this can be confusing.

What is the reason for not simply calling this Authenticator?

Reply:
Here the main reason is that, I am simply following the ARM and Admin authorizer implemented before the mise one. Also, the capability of this authorizer is not limited to authentication but rather role based authorizing using actor claims as well which can be leveraged.

Reply:
I understand the prior work here that makes this a little bit confusing.

@sankur-codes I think we should create a Jira for this and iterate later. 

As a good practice, the code should make it clear whether it is authorizing or authenticating only or both. It's a good practice because it enhances readability, it tells developers where and what the code is actually doing. For example, if I want to find and understand the authorization methods of this service, then I can find it easily and focus on it. In this Authorizer's case, if not careful enough the fact that it's not authorizing and only authentication can be missed easily.

Reply:
+1 to handling this change later.